cmd/system-probe/modules/ping.go

```diff
@@ -0,0 +1,142 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux
+
+package modules
+
+import (
+	"encoding/json"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/DataDog/datadog-agent/cmd/system-probe/api/module"
+	"github.com/DataDog/datadog-agent/cmd/system-probe/config"
+	sysconfigtypes "github.com/DataDog/datadog-agent/cmd/system-probe/config/types"
+	pingcheck "github.com/DataDog/datadog-agent/pkg/networkdevice/pinger"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/gorilla/mux"
+	"go.uber.org/atomic"
+	"google.golang.org/grpc"
+)
+
+const (
+	countParam    = "count"
+	intervalParam = "interval"
+	timeoutParam  = "timeout"
+)
+
+type pinger struct{}
+
+// Pinger is a factory for NDMs Ping module
+var Pinger = module.Factory{
+	Name:             config.PingModule,
+	ConfigNamespaces: []string{},
+	Fn: func(cfg *sysconfigtypes.Config) (module.Module, error) {
+		return &pinger{}, nil
+	},
+	NeedsEBPF: func() bool {
+		return false
+	},
+}
+
+var _ module.Module = &pinger{}
+
+func (p *pinger) GetStats() map[string]interface{} {
+	return nil
+}
+
+func (p *pinger) Register(httpMux *module.Router) error {
+	var runCounter = atomic.NewUint64(0)
+
+	httpMux.HandleFunc("/ping/{host}", func(w http.ResponseWriter, req *http.Request) {
+		start := time.Now()
+		vars := mux.Vars(req)
+		id := getClientID(req)
+		host := vars["host"]
+
+		count, err := getIntParam(countParam, req)
```

Comment:
Do we have sane defaults for when `getIntParam` returns `0`?

Reply:
Yes, [the underlying function](https://github.com/DataDog/datadog-agent/pull/21939/files#diff-531760b7f55b3c10628eb8adfaf8fdf7ae42839950cb3081d018007cba2b6106R22-R34) checks to ensure none of these are 0.

Reply:
If one of the query parameters is missing, we'd still get a `0` right?